build_mmdeploy.sh

```diff
@@ -0,0 +1,346 @@
+#!/bin/bash
+# build_mmdeploy.sh
+#   Date: 08-03-2022, 24-04-2022
+#
+#   Run this script to build MMDeploy SDK and prerequisites.
+#   This script will also setup python venv
+#
+BUILD_TYPE="Release"
+ARCH=$(uname -i)
+WITH_PYTHON=1
+WITH_CLEAN=1
+# Get path to script to ensure script runs from 'logends' root
+WORKING_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+PPLCV_DIR=${WORKING_DIR}/ppl.cv
+MMDEPLOY_DIR=${WORKING_DIR}/MMDeploy
+PPLCV_VER="0.6.2"
+MMDEPLOY_VER="0.4.0"
+INSTALL_PREFIX="/usr/local"
+PYTHON_VENV_DIR="mmdeploy"
+
+appargument1=$1
+#appargument2=$2
+
+echo_green() {
+  if [ -n "$1" ]; then
+    echo "$(tput setaf 10)$1$(tput sgr 0)"
+  fi
+}
+echo_red() {
+  if [ -n "$1" ]; then
+    echo "$(tput setaf 1)$1$(tput sgr 0)"
+  fi
+}
+echo_blue() {
+  if [ -n "$1" ]; then
+    echo "$(tput setaf 4)$1$(tput sgr 0)"
+  fi
+}
+contains_element () {
+  local e match="$1"
+  shift
+  for e; do [[ "$e" == "$match" ]] && return 0; done
+  return 1
+}
+
+prereqs() {
+  # spdlog
+  read -p "Install latest spdlog from source? (y/n)" -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]
+  then
+    # remove libspdlog, as it might be an old version
+    sudo apt-get remove libspdlog-dev -y
+
+    echo "Building and installing latest spdlog from source"
+    git clone https://github.com/gabime/spdlog.git spdlog
+    cd spdlog
+    git pull
+    git checkout tags/v1.8.1
+    mkdir build -p && cd build
+    # we must build spdlog with -fPIC enabled
+    cmake .. -DCMAKE_POSITION_INDEPENDENT_CODE=ON && make -j
+    sudo make install
+    sudo ldconfig
+  fi
+
+  # cmake check
+  echo_green "Check your cmake version:"
+  cmake --version
+  read -p "Install latest CMake? (>=3.18 is required) (y/n)" -n 1 -r
```

Comment:
I think it's better to install `cmake` automatically when its version is less than `v3.14`.
And [CMake](https://cmake.org/download/) already provides the prebuilt binary. For example, 

Linux x86_64 | [cmake-3.23.1-linux-x86_64.sh](https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1-linux-x86_64.sh)
-- | --
  | [cmake-3.23.1-linux-x86_64.tar.gz](https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1-linux-x86_64.tar.gz)
Linux aarch64 | [cmake-3.23.1-linux-aarch64.sh](https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1-linux-aarch64.sh)
  | [cmake-3.23.1-linux-aarch64.tar.gz](https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1-linux-aarch64.tar.gz)

Therefore, instead of building it from the source code, we can download the binary. That would save quite an amount of time.

Reply:
@lvhan028 Indeed, the CMake build is slow (I previously used snap install as an alternative but changed it back to building from source). This looks like a cleaner solution so I will change it. It is also a good idea to do this without user interaction.

Reply:
If an older version of cmake is detected, we should probably purge it and install the latest(?) to the user folder as listed below:

```
# purge existing
sudo apt-get purge cmake
sudo snap remove cmake
#
wget https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1-linux-${ARCH}.sh
chmod +x cmake-3.23.1-linux-${ARCH}.sh
sudo ./cmake-3.23.1-linux-${ARCH}.sh --prefix=/usr --skip-license
```

This is slightly different than the method described in the build documentation for MMDeploy, as we use symlinks to /usr/bin instead of running the install script.

This method should work irrespective of Ubuntu version and we avoid snap packages.